memphis/producer.py

```diff
@@ -232,6 +235,11 @@ async def produce(
             else:
                 headers = memphis_headers
 
+            if not self.internal_station_name in self.connection.partition_producers_updates_data:
```

Comment:
Change to `if self.internal_station_name not in self.connection.partition_producers_updates_data`